sktime/transformations/_delegate.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+"""Delegator mixin that delegates all methods to wrapped transformer.
+
+Useful for building estimators where all but one or a few methods are delegated.
+For that purpose, inherit from this estimator and then override only the methods
+    that are not delegated.
+"""
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+__author__ = ["miraep8"]
+__all__ = ["_DelegatedTransformer"]
+
+from sktime.transformations.base import BaseTransformer
+
+
+class _DelegatedTransformer(BaseTransformer):
+    """Delegator mixin that delegates all methods to wrapped transformer.
+
+    Delegates inner transformer methods to a wrapped estimator.
+        Wrapped estimator is value of attribute with name self._delegate_name.
+        By default, this is "estimator_", i.e., delegates to self.estimator_
+        To override delegation, override _delegate_name attribute in child class.
+
+    Delegates the following inner underscore methods:
+        _fit, _transform, _inverse_transform, _update
+
+    Does NOT delegate get_params, set_params.
+        get_params, set_params will hence use one additional nesting level by default.
+
+    Does NOT delegate or copy tags, this should be done in a child class if required.
+    """
+
+    _delegate_name = "estimator_"
+
+    def _get_delegate(self):
+        return getattr(self, self._delegate_name)
+
+    def _fit(self, X, y=None):
+        """Fit transformer to X, optionally to y.
+
+        State change:
+            Changes state to "fitted".
+
+        Writes to self:
+            Sets is_fitted flag to True.
```

Comment:
non-blocking issue - this is not correct. The setting of the `is_fitted` flag is done by `fit`, not `_fit`.